src/dune_rules/warning_emit.ml

```diff
@@ -0,0 +1,31 @@
+open Import
+open Memo.O
+
+type context =
```

Comment:
Can we use a more descriptive name that `context` here? I don't have a clear sense of how to choose the `context` argument to `emit`. Is the idea that every warning is because of something the user wrote in a dune{,-project,-workspace} file and that the path to that file (or maybe the directory containing the file?) will determine whether or not it's in a vendored project?

Reply:
Logically, a context is what decides if a particular warning is active. E.g, we can use it in the following `val warning_active -> Warning.t -> context -> bool Memo.t`.

Reply:
Right but that doesn't help me choose which path to use for the context of a particular warning. If I want to raise a warning and there are multiple paths in scope which one should I use. If there's only one path in scope how do I know it's the right one for the warning?

Reply:
It should be clear now.